File: my-folder/0957-minimum-add-to-make-parentheses-valid/solution.java
class Solution {
    public int minAddToMakeValid(String S) {
        int left = 0;
        int right = 0;
        for (char c : S.toCharArray()) {
            if (c == '(') {
                right++;
            } else {
                if (right > 0) {
                    right--;
                } else {
                    left++;
                }
            }
        }
        return left + right;
    }
}

File: my-folder/0957-minimum-add-to-make-parentheses-valid/solution.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        need = 0
        res = 0
        for i in range(len(s)):
            if s[i] == '(':
                need += 1
            if s[i] == ')':
                if need > 0:
                    need -= 1
                else:
                    res += 1
        return res + need




